src/main/java/calculator/domain/StringCalculator.java

```diff
@@ -9,27 +9,26 @@
 
 public class StringCalculator {
 
-    private static final String CUSTOM_DELIMITER_PATTERN = "//(.)\n(.*)";
+    private static final Pattern CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\n(.*)");
     private String delimiter = ",|:";
 
-
     public StringCalculator() {
     }
 
     public int add(String inputString) {
-        if(StringUtils.isEmpty(inputString)) return 0;
+        if(StringUtils.isEmpty(inputString)) {
+            return 0;
+        }
 
         String[] tokens = getTokens(inputString);
 
         return Arrays.stream(tokens)
-                .peek(token -> {
-                    if (isNotNumber(token) || isNegativeNumber(token)) throw new RuntimeException("음수 또는 문자열은 계산할 수 없습니다.");
-                })
-                .collect(Collectors.summingInt(Integer::parseInt));
+                .map(token -> new CalculatorNumber(token))
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
                .map(CalculatorNumber::new)
```

Reply:
알려주신 방법으로 수정하니 가독성이 더 좋아진것 같습니다. :)